### PR TITLE
Create compute target on behalf of user

### PR DESCRIPTION
**Pull request by FredFLi**
1. Create compute target on behalf of user #101
2. Hide printing of action by default #100

### CODE REVIEW

**Comment by lostmygithubaccount** [comment]
@jacobtomlinson this simplifies the creation a lot

File: doc/source/index.rst [hunk 1]
```diff
@@ -233,134 +228,21 @@ You can use ``ws = Workspace.from_config()`` after downloading the config file f
...
-   )
+   # common optional kwargs
+   vm_size = "STANDARD_DS13_V2"                       # Azure VM size for the Compute Target
+   datastores = ws.datastores.values()                # Azure ML Datastores to mount on the headnode
+   environment = ws.environments['AzureML-Dask-CPU]   # Azure ML Environment to run on the cluster_ar
+   initial_node_count = 20                            # number of nodes to start
+   scheduler_idle_timeout = 7200                      # scheduler idle timeout in seconds 
+
+   # only workspace is required
+   amlcluster = AzureMLCluster(ws)
```
Comment (jacobtomlinson) [inline doc/source/index.rst:245]: Could we leave this with the Python example, but maybe show the optional arguments clearly. Makes copy/paste easier for users.

```suggestion
   amlcluster = AzureMLCluster(
       # Required kwargs
       workspace=ws,
       
       # Optional kwargs
       vm_size="STANDARD_DS13_V2",                        # Azure VM size for the Compute Target
       datastores=ws.datastores.values(),                 # Azure ML Datastores to mount on the headnode
       environment=ws.environments['AzureML-Dask-CPU],    # Azure ML Environment to run on the cluster_ar
       initial_node_count=20,                             # number of nodes to start
       scheduler_idle_timeout 7200                        # scheduler idle timeout in seconds 
    )
```

File: dask_cloudprovider/providers/azure/azureml.py [hunk 2]
```diff
@@ -52,7 +67,7 @@ class AzureMLCluster(Cluster):
     jupyter: bool (optional)
         Flag to start JupyterLab session on the headnode of the cluster.
 
-        Defaults to ``False``.
+        Defaults to ``True``.
```
Comment (jacobtomlinson) [inline dask_cloudprovider/providers/azure/azureml.py:70]: This should be `False`

File: dask_cloudprovider/providers/azure/azureml.py [hunk 3]
```diff
@@ -514,23 +655,27 @@ async def __setup_port_forwarding(self):
                     f"setsid socat tcp-listen:{self.port[1]},reuseaddr,fork tcp:{scheduler_ip}:{port[0]} &"
                 )
         else:
+            forwarding_option = "L"
```
Comment (jacobtomlinson) [inline dask_cloudprovider/providers/azure/azureml.py:658]: What is the purpose of moving this out into a variable?

Comment (FredFLi) [reply dask_cloudprovider/providers/azure/azureml.py:658]: Thanks, Jacob. No specific purpose. I was testing remote forwarding with -R. I can change it back if this is a concern.

Comment (jacobtomlinson) [reply dask_cloudprovider/providers/azure/azureml.py:658]: Yeah, I would prefer to reduce complexity as much as possible. It just makes the code a little harder to parse when coming back to it.

Comment (FredFLi) [reply dask_cloudprovider/providers/azure/azureml.py]: Updated. Thanks.

**Comment by lostmygithubaccount** [comment]
@jacobtomlinson have you had a chance to test? I think all the comments have been addressed

**Comment by jacobtomlinson** [comment]
I haven't yet sorry. Will try and get to it tomorrow.

**Comment by lostmygithubaccount** [comment]
@jacobtomlinson will you be able to test and merge soon?

**Comment by jacobtomlinson** [comment]
Sorry, thanks for the nudging. Merging now!

**Pull request merged**